protocol/identity/alias/generate.go

```diff
@@ -32,16 +29,27 @@ func GenerateFromPublicKey(publicKey *ecdsa.PublicKey) string {
 // GenerateFromPublicKeyString returns the 3 words name given a public key
 // prefixed with 0x
 func GenerateFromPublicKeyString(publicKeyString string) (string, error) {
-	pk := strings.TrimPrefix(publicKeyString, "0x")
-	publicKeyBytes, err := hex.DecodeString(pk)
-	if err != nil {
-		return "", err
+	var fixedPublicKey string
+	if len(publicKeyString) == multiformat.LegacyKeyLength {
+		fixedPublicKey = publicKeyString
+	} else {
+		// Attempt to fix a maybe public key without the 0x prefix
+		if len(publicKeyString) == multiformat.LegacyKeyLength-2 {
```

Comment:
This code is needed to not break the API of the previous implementation, that implementation accepted public keys without the prefix `0x`.

Reply:
Could we check the first two characters are not `0x` instead of only checking the length? Seems to be a bit more robust.

Reply:
Yes, a lot more robust, thanks! I'll fix it.